### PR TITLE
Feature: iOS rectOfInterest implementation

### PR DESCRIPTION
**Pull request by chubakueno**
Implementation of iOS [rectOfInterest](https://developer.apple.com/documentation/avfoundation/avcapturemetadataoutput/1616291-rectofinterest) for barcode reading

### CODE REVIEW

File: ios/RN/RNCamera.m [hunk 1]
```diff
@@ -588,7 +589,7 @@ - (void)startSession
         }
 
         self.session.sessionPreset = AVCaptureSessionPresetPhoto;
-
+
```
Comment (n1ru4l) [inline ios/RN/RNCamera.m]: Could you please remove all the padding you added?

File: src/RNCamera.js [hunk 2]
```diff
@@ -411,6 +445,8 @@ export default class Camera extends React.Component<PropsType, StateType> {
           onPictureSaved={this._onPictureSaved}
         >
           {this.renderChildren()}
+          {this.props.rectOfInterest !== undefined &&
```
Comment (sibelius) [inline src/RNCamera.js]: I think drawing rect of interesting should be in user land, instead of here

can you remove it?

**Comment by sibelius** [comment]
@chubakueno  can you rebase?

do you have the same on android?

**Comment by andresmtz98** [comment]
Any update?

**Comment by sibelius** [comment]
we need someone to rebase

**Comment by chubakueno** [comment]
Everything has been rebased to a nice commit on top of master HEAD.

File: src/RNCamera.js (hunk 2 above)
Comment (chubakueno) [reply src/RNCamera.js]: Done

File: src/RNCamera.js [hunk 3]
```diff
@@ -401,6 +409,7 @@ export default class Camera extends React.Component<PropsType, StateType> {
     videoStabilizationMode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
     pictureSize: PropTypes.string,
     mirrorVideo: PropTypes.bool,
+    rectOfInterest: PropTypes.any,
```
Comment (n1ru4l) [inline src/RNCamera.js]: Please use `PropTypes.shape`

**Comment by fabriziobertoglio1987** [comment]
https://github.com/react-native-community/react-native-camera/pull/2766

**Pull request merged**